io.github.nyre221.kiview.json

```diff
@@ -0,0 +1,45 @@
+{
+   "base": "io.qt.qtwebengine.BaseApp",
+   "base-version": "6.6",
+   "command": "kiview",
+   "build-options": {
+    "env": [
+      "QTWEBENGINE_DISABLE_SANDBOX=1"
+    ]
+  },
+
+   "finish-args": [
+      "--share=ipc",
+      "--device=dri",
+      "--socket=fallback-x11",
+      "--socket=wayland",
+      "--env=QML_IMPORT_PATH=/app/qml" ,
+      "--socket=pulseaudio",
+      "--socket=session-bus",
+      "--filesystem=host:ro",
+      "--filesystem=/tmp",
+      "--talk-name=org.freedesktop.Flatpak",
```

Comment:
this needs an exception.

Reply:
what do you mean?

Reply:
https://docs.flathub.org/docs/for-app-authors/linter#exceptions

Reply:
What commands from libreoffice do you need? unrar can be easily included in the flatpak here.

Reply:
libreoffice --headless --nolockcheck --norestore --convert-to pdf file.[doc,ppt,etc,etc] --outdir /tmp

And I need to kill the process with `kill` if the user switches to another file.

How do I include unrar?

Reply:
you mean the open source version of unrar, right? I remember trying to use that version and it fails to open some .rar archives.

Reply:
> How do I include unrar?

https://github.com/search?q=org%3Aflathub%20unrar&type=code

> kill if the user switches to another file.

`kill` is available inside the sandbox.

> libreoffice --headless --nolockcheck --norestore --convert-to pdf file.[doc,ppt,etc,etc] --outdir /tmp

Libreoffice provides some prebuild binary packages, can you try to see if including them in the flatpak works? https://www.libreoffice.org/download/appimage/ or the deb/rpm https://www.libreoffice.org/download/download-libreoffice/

This is not very ideal but still better than having a gaping hole.

Reply:
And if it is included in Flatpak, it works ootb for people.

You can try setting `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"` or to `/var/tmp` to drop the `filesystem=/tmp` access.

Reply:
I'll test it tomorrow and let you know.

I'll probably remove support for  .rar files if the version to include is the open source one that doesn't always work.

Thanks for your help.

Reply:
I managed to use /var/tmp:ro and removed /tmp, unfortunately I still haven't managed to "completely" start libreoffice in the sandbox (Appimage). Everything works except libreoffice-impress which gives me the error:

![Schermata_20240407_221344](https://github.com/flathub/flathub/assets/104171042/ee59c1e0-fb76-4797-9748-71be0e7dd4c0)

I can try with the .deb but I don't think it will change anything.

In any case I no longer need access to /tmp

I was thinking, is there any way to extend another application similarly to what I did with WebEngine?
I tried copying most of the libreoffice .json and couldn't get it to install.

I also included unrar.

Reply:
@bbhtt 
Even installing openjdk didn't fix the error. 
I don't know what else to try.

Reply:
I'll check later.

Reply:
@bbhtt 
I think I found the problem:
https://bugs.documentfoundation.org/show_bug.cgi?id=119408
I'll see what I can do tomorrow.

Reply:
I think I fixed the error.
If everything now works as I think, the only problematic permission is session-bus.

Reply:
We can make an exception for session bus since there is no other way, please make a PR for the session-bus exception. https://docs.flathub.org/docs/for-app-authors/linter/#finish-args-arbitrary-dbus-access https://docs.flathub.org/docs/for-app-authors/linter#exceptions

Reply:
It's because the name ends with `-{id}`, the `*` wildcard only works when ending with `.*`, it's not a generic wildcard that expands everywhere.

The best you can do is `org.kde.*` probably, that's somewhat better than having a session-bus permission

You'll still need to make an exception request as that is blocked too
https://docs.flathub.org/docs/for-app-authors/linter#finish-args-wildcard-kde-talk-name